lib/HLSL/DxilEliminateOutputDynamicIndexing.cpp

```diff
@@ -0,0 +1,163 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilCondenseResources.cpp                                                 //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Eliminate dynamic indexing on output.                                     //
```

Comment:
I think we need a more general approach here. The goal is provide a pass that will emulate the previous compiler behavior, which means we should not also allow stores to outputs multiple times. What if we create an alloca for each output and demote the storeOutput calls to writes to the alloca. Then add a store output that stores the final value from the alloca.